lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs

```diff
@@ -157,22 +157,58 @@ public void DeleteDirectory(Uri directoryUri)
         }
 
         /// <summary>
-        /// Gets the children of the blob "directory."
+        /// Gets the children of the container (if the uri has segments)
+        /// or containers in the storage account (if the uri has no segments)
         /// </summary>
         public IEnumerable<Uri> GetChildren(Uri directoryUri)
         {
+            string[] segments = directoryUri.Segments;
             BlobContinuationToken blobContinuationToken = null;
-            var path = directoryUri.AbsolutePath.Trim('/');
+
+            // If at the root, return all containers
+            if (segments.Count() == 1)
+            {
+                do
+                {
+                    ContainerResultSegment containerListing = _client.ListContainersSegmented(blobContinuationToken);
+
+                    if (containerListing.Results != null)
+                    {
+                        foreach (CloudBlobContainer containerItem in containerListing.Results)
+                        {
+                            yield return containerItem.Uri;
+                        }
+                    }
+
+                    blobContinuationToken = containerListing.ContinuationToken;
+                }
+                while (blobContinuationToken != null);
+                yield break;
+            }
+
+            // If not at the root folder, return all blobs within the container
+            string containerName = segments[1];
```

Comment:
Possible out of bounds exception.

Reply:
Fixed by changing the condition earlier by ensuring this line is not executed if segments.Count() <= 1.